src/dlls/mscoree/unixinterface.cpp

```diff
@@ -299,6 +299,39 @@ int coreclr_shutdown(
 }
 
 //
+// Shutdown CoreCLR. It unloads the app domain and stops the CoreCLR host.
+//
+// Parameters:
+//  hostHandle              - Handle of the host
+//  domainId                - Id of the domain
+//  latchedExitCode         - Latched exit code after domain unloaded
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the assembly was successfully executed
+//
+extern "C"
+int coreclr_shutdown_2(
+            void* hostHandle,
+            unsigned int domainId,
+            int* latchedExitCode)
+{
+    ICLRRuntimeHost4* hostHandleV4 = nullptr;
+    reinterpret_cast<ICLRRuntimeHost2*>(hostHandle)->QueryInterface(IID_ICLRRuntimeHost4, reinterpret_cast<void**>(&hostHandleV4));
```

Comment:
Instead of this query interface here, could you please change all references in this file from ICLRRuntimeHost2 to ICLRRuntimeHost4?